libcontainer/rootfs_linux.go

```diff
@@ -215,6 +215,17 @@ func prepareRootfs(pipe *syncSocket, iConfig *initConfig) (err error) {
 		return fmt.Errorf("error jailing process inside rootfs: %w", err)
 	}
 
+	// Apply root mount propagation flags.
+	// This must be done after pivot_root/chroot because the mount propagation flag is applied
+	// to the current root ("/"), and not to the old rootfs before it becomes "/". Applying the
+	// flag in prepareRoot would affect the host mount namespace if the container's
+	// root mount is shared.
+	if config.RootPropagation != 0 {
```

Comment:
I think if we add a `unix.MS_PRIVATE` comparison, it will reduce a mount syscall.
```
if config.RootPropagation != 0 && config.RootPropagation & unix.MS_PRIVATE == 0 {
```

Reply:
fixed.